lib/colors.nix

```diff
@@ -69,57 +80,204 @@ let
         g = builtins.substring 2 2 hex;
         b = builtins.substring 4 2 hex;
       };
-      splitRGB' = f: hex: prefix:
-        let rgb = splitRGB hex;
-        in {
-          "${prefix}-r" = f rgb.r;
-          "${prefix}-g" = f rgb.g;
-          "${prefix}-b" = f rgb.b;
-        };
-      addRGB = f: prefix:
-        builtins.foldl' (x: y: x // y) { }
-        (builtins.map (baseXX: splitRGB' f base.${baseXX} "${baseXX}-${prefix}")
-          (builtins.attrNames base));
+      inherit (builtins) listToAttrs attrValues;
+      uncurry = lib.foldl lib.id;
+      mapAttrValues = compose [
+        lib.const
+        lib.mapAttrs
+      ];
+      _color =
+        hex:
+        lib.fix (self: {
+          hex = {
+            __toString = self: hex;
+            inherit (splitRGB self.hex) r g b;
+            bgr = with self.hex; b + g + r;
+            withHashtag = "#${self.hex}";
+          };
+          __toString = self: self.hex;
+          rgb = mapAttrValues toIntBase16 (splitRGB self.hex);
+          dec = mapAttrValues (x: x / 255.0) self.rgb;
+        });
 
-      # populate the fields:
-      base-hex-rgb = addRGB (x: x) "hex";
-      base-rgb-rgb = addRGB (x: builtins.toString (primaryHex2Dec x)) "rgb";
-      base-dec-rgb = addRGB (x: builtins.toString (primaryHex2Dec x / 256.0)) "dec";
+      ansi-list =
+        colors: with colors; [
+          black
+          red
+          green
+          yellow
+          blue
+          magenta
+          cyan
+          white
+        ];
+      ansi =
+        palette:
+        with mkBase24 palette;
+        lib.fix (self: {
+          dark = {
+            black = base01;
+            red = base08;
+            green = base0B;
+            yellow = base09;
+            blue = base0D;
+            magenta = base0E;
+            cyan = base0C;
+            white = base06;
+            # suitable for programs.wezterm.colorSchemes.<name>.ansi and similar
+            toList = ansi-list self.dark;
+          };
+          bright = {
+            black = base02;
+            red = base12;
+            green = base14;
+            yellow = base13;
+            blue = base16;
+            magenta = base17;
+            cyan = base15;
+            white = base07;
+            toList = ansi-list self.bright;
+          };
+          # suitable for console.colors and similar
+          toList = self.dark.toList ++ self.bright.toList;
```

Comment:
This breaks the backward compatibility (order is mixed up)

Reply:
This is a new attribute, it does not change any order as it did not exist prior to this change